### ISSUE TITLE
rate limiting etc

### ISSUE THREAD

**Issue by gedw99** [dkotik/htadaptor#1]
Hey 

Really clean package. 

Would you be open to PR's that add rate limiting and other gateway patterns ?

**Comment by dkotik** [comment]
Yes! Right now everything is dumped into the same Go module, but I intend to add `go.mod` to either middleware folder or individual middlewares in order to reduce core dependencies to standard library only.

Regarding rate limiting:

1. Seth Vargo's <https://github.com/sethvargo/go-limiter> will work with htadaptor no problem, because they are just vanilla http.HandlerFuncs.
2. I was doodled some improvements on Seth's approach here: <https://github.com/dkotik/oakratelimiter> (work in progress)

Your ideas and contributions would be welcome - eager to cooperate.

**Comment by gedw99** [comment]
cool.

There are some nice ones here: https://github.com/zeromicro/go-zero/tree/master/core

all have zero dependencies, just like htadaptor.

SO maybe we can draw inspiration from these.

Have a look and see what you think...

**Comment by dkotik** [comment]
An interesting throttling library that could be shimmed in: https://github.com/1pkg/gohalt
